File: app/src/main/java/com/legue/axel/lolsummonertool/database/viewmodel/ItemViewModel.java
package com.legue.axel.lolsummonertool.database.viewmodel;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.annotation.NonNull;

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase;
import com.legue.axel.lolsummonertool.database.model.item.Item;
import com.legue.axel.lolsummonertool.database.model.item.ItemImage;

import java.util.List;

public class ItemViewModel extends AndroidViewModel {

    private LiveData<List<Item>> items;
    private SummonerToolDatabase database;

    public ItemViewModel(@NonNull Application application) {
        super(application);
        database = SummonerToolDatabase.Companion.getInstance(this.getApplication());
        items = database.itemDao().getItems();
    }

    public LiveData<List<Item>> getItems() {
        return items;
    }

    public LiveData<Item> getItemById(int itemId) {
        return database.itemDao().getItemById(itemId);
    }

    public LiveData<ItemImage> getItemImage(int itemId) {
        return database.itemImageDao().getItemImageByItemId(itemId);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/viewmodel/ItemViewModel.kt
package com.legue.axel.lolsummonertool.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.legue.axel.lolsummonertool.database.SummonerToolDatabase
import com.legue.axel.lolsummonertool.database.model.item.Item
import com.legue.axel.lolsummonertool.database.model.item.ItemGold
import com.legue.axel.lolsummonertool.database.model.item.ItemImage
import com.legue.axel.lolsummonertool.repository.ItemRepository

class ItemViewModel(application: Application) : AndroidViewModel(application) {

    fun start(itemId: Int) {
        id = itemId
    }

    private val itemRepository: ItemRepository = ItemRepository.getInstance(application)
    private var id: Int = 0

    fun getItems(): LiveData<List<Item>>  = itemRepository.getItems()
    fun getItemById(): LiveData<Item> = itemRepository.getItemById(id)
    fun getItemImage(): LiveData<ItemImage> = itemRepository.getItemImage(id)
    fun getItemGoldByItemId(): LiveData<ItemGold> = itemRepository.getItemGoldByItemId(id)

}
